### PR TITLE
Refactor test

### PR DESCRIPTION
**Pull request by ajax-marchenko-s**
- change test naming
- change test library (from mockito to mockk)

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -32,7 +33,7 @@ dependencies {
     testImplementation("org.jetbrains.kotlin:kotlin-test-junit5")
     testRuntimeOnly("org.junit.platform:junit-platform-launcher")
     testImplementation("org.springframework.security:spring-security-test")
-    testImplementation ("org.mockito.kotlin:mockito-kotlin:3.2.0")
+    testImplementation("io.mockk:mockk:1.13.8")
 }
```
Comment (ajax-yurkovskyi-t) [inline build.gradle.kts:37]: change to last stable version

File: src/test/kotlin/ua/marchenko/artauction/auction/service/AuctionServiceTest.kt [hunk 2]
```diff
@@ -12,25 +12,40 @@ import ua.marchenko.artauction.auction.repository.AuctionRepository
...
+import org.junit.jupiter.api.extension.ExtendWith
 import kotlin.test.Test
-import org.mockito.Mockito.mock
-import org.mockito.kotlin.doReturn
-import org.mockito.kotlin.whenever
 import ua.marchenko.artauction.artwork.model.Artwork
 import ua.marchenko.artauction.auction.controller.dto.CreateAuctionRequest
 import ua.marchenko.artauction.auction.model.Auction
 
+@ExtendWith(MockKExtension::class)
```
Comment (ajax-yurkovskyi-t) [inline src/test/kotlin/ua/marchenko/artauction/auction/service/AuctionServiceTest.kt:27]: u can remove this

**Review by ajax-yurkovskyi-t — COMMENTED** [review]
Good job!

File: build.gradle.kts (hunk 1 above)
Comment (ajax-marchenko-s) [reply build.gradle.kts]: done

File: src/test/kotlin/ua/marchenko/artauction/auction/service/AuctionServiceTest.kt (hunk 2 above)
Comment (ajax-marchenko-s) [reply src/test/kotlin/ua/marchenko/artauction/auction/service/AuctionServiceTest.kt]: done

File: src/test/kotlin/ua/marchenko/artauction/artwork/controller/ArtworkControllerTest.kt [hunk 3]
```diff
@@ -9,24 +9,33 @@ import ua.marchenko.artauction.artwork.service.ArtworkService
...
+    @MockK
+    private lateinit var mockArtworkService: ArtworkService
+
+    @InjectMockKs
+    private lateinit var artworkController: ArtworkController
+
+    @BeforeEach
+    fun setUp() {
+        MockKAnnotations.init(this)
+    }
```
Comment (ajax-shulepa-a) [inline src/test/kotlin/ua/marchenko/artauction/artwork/controller/ArtworkControllerTest.kt:32]: there should be way to enable annotations by default
maybe this property would help you
systemProperty("junit.jupiter.extensions.autodetection.enabled", "true")

https://junit.org/junit5/docs/current/user-guide/#extensions-registration-automatic

File: src/test/kotlin/ua/marchenko/artauction/artwork/controller/ArtworkControllerTest.kt [hunk 4]
```diff
@@ -9,24 +9,33 @@ import ua.marchenko.artauction.artwork.service.ArtworkService
...
 
     @Test
-    fun `getAllArtworks should return a list of ArtworkResponse`() {
+    fun `should return a list of ArtworkResponse when there are some artworks`() {
         // GIVEN
         val artworks = listOf(Artwork.random())
-        whenever(mockArtworkService.getAll()) doReturn (artworks)
+        every { mockArtworkService.getAll() } returns (artworks)
```
Comment (ajax-shulepa-a) [inline src/test/kotlin/ua/marchenko/artauction/artwork/controller/ArtworkControllerTest.kt:38]:
```suggestion
        every { mockArtworkService.getAll() } returns artworks
```
fix in all places

File: src/test/kotlin/ua/marchenko/artauction/artwork/controller/ArtworkControllerTest.kt [hunk 5]
```diff
@@ -37,9 +46,9 @@ class ArtworkControllerTest {
     }
 
     @Test
-    fun `getAllArtworks should return an empty list if there are no artworks`() {
+    fun `should return an empty list when there are no artworks`() {
         // GIVEN
-        whenever(mockArtworkService.getAll()) doReturn (listOf())
+        every { mockArtworkService.getAll() } returns (listOf())
```
Comment (ajax-shulepa-a) [inline src/test/kotlin/ua/marchenko/artauction/artwork/controller/ArtworkControllerTest.kt:51]:
```suggestion
        every { mockArtworkService.getAll() } returns emptyList()
```
emptyList has more descriptive name

File: src/test/kotlin/ua/marchenko/artauction/auction/service/AuctionServiceTest.kt [hunk 6]
```diff
@@ -68,31 +81,31 @@ class AuctionServiceTest {
...
     }
 
     @Test
-    fun `save should change artwork status and save`() {
+    fun `should change artwork status and save when artwork exist and artwork status is view`() {
         //GIVEN
         val newAuctionId = getRandomObjectId()
-        val artwork = Artwork.random()
+        val artwork = Artwork.random(status = ArtworkStatus.VIEW)
         val auctionRequest = CreateAuctionRequest.random(artworkId = artwork.id!!.toString())
```
Comment (ajax-shulepa-a) [inline src/test/kotlin/ua/marchenko/artauction/auction/service/AuctionServiceTest.kt:98]:
```suggestion
        val auctionRequest = CreateAuctionRequest.random(artworkId = artwork.id!!.toHexString())
```

File: src/test/kotlin/ua/marchenko/artauction/auth/service/AuthServiceTest.kt [hunk 7]
```diff
@@ -5,77 +5,85 @@ import org.junit.jupiter.api.Assertions.assertEquals
...
-        whenever(mockJwtService.generate(userDetails)) doReturn (expectedResponse.accessToken)
+        every { mockAuthManager.authenticate(usernamePasswordAuthenticationToken) } returns (null)
+        every { mockUserDetailsService.loadUserByUsername(authenticationRequest.email) } returns (userDetails)
+        every { mockJwtService.generate(userDetails) } returns (expectedResponse.accessToken)
 
         //WHEN
         val result = authenticationService.authentication(authenticationRequest)
 
         //THEN
-        verify(mockAuthManager).authenticate(usernamePasswordAuthenticationToken)
```
Comment (ajax-shulepa-a) [inline src/test/kotlin/ua/marchenko/artauction/auth/service/AuthServiceTest.kt:60]: instead of deleting verification, replace it with mockk verification

**Review by ajax-shulepa-a — CHANGES_REQUESTED** [review]

File: src/test/kotlin/ua/marchenko/artauction/artwork/controller/ArtworkControllerTest.kt (hunk 4 above)
Comment (ajax-marchenko-s) [reply src/test/kotlin/ua/marchenko/artauction/artwork/controller/ArtworkControllerTest.kt]: done

File: src/test/kotlin/ua/marchenko/artauction/artwork/controller/ArtworkControllerTest.kt (hunk 5 above)
Comment (ajax-marchenko-s) [reply src/test/kotlin/ua/marchenko/artauction/artwork/controller/ArtworkControllerTest.kt]: done

File: src/test/kotlin/ua/marchenko/artauction/auction/service/AuctionServiceTest.kt (hunk 6 above)
Comment (ajax-marchenko-s) [reply src/test/kotlin/ua/marchenko/artauction/auction/service/AuctionServiceTest.kt]: done

File: src/test/kotlin/ua/marchenko/artauction/auth/service/AuthServiceTest.kt (hunk 7 above)
Comment (ajax-marchenko-s) [reply src/test/kotlin/ua/marchenko/artauction/auth/service/AuthServiceTest.kt:60]: done

**Review by ajax-shulepa-a — APPROVED** [review]

**Pull request merged**